vm/src/hint_processor/builtin_hint_processor/bootloader/program_loader.rs

```diff
@@ -0,0 +1,332 @@
+use crate::hint_processor::builtin_hint_processor::bootloader::types::BootloaderVersion;
+use crate::serde::deserialize_program::BuiltinName;
+use crate::types::program::Program;
+use crate::types::relocatable::Relocatable;
+use crate::vm::errors::hint_errors::HintError;
+use crate::vm::errors::memory_errors::MemoryError;
+use crate::vm::vm_memory::memory::Memory;
+use felt::Felt252;
+
+#[derive(thiserror_no_std::Error, Debug)]
+pub enum ProgramLoaderError {
+    #[error(transparent)]
+    Memory(#[from] MemoryError),
+
+    #[error("Program has no entrypoint")]
+    NoEntrypoint,
+}
+
+impl Into<HintError> for ProgramLoaderError {
+    fn into(self) -> HintError {
+        match self {
+            ProgramLoaderError::Memory(e) => HintError::Memory(e),
+            _ => HintError::CustomHint(self.to_string().into_boxed_str()),
+        }
+    }
+}
+
+/// Creates an instance of `Felt252` from a builtin name.
+///
+/// Converts the builtin name to bytes then attempts to create a felt from
+/// these bytes. This function will fail if the builtin name is over 31 characters.
```

Comment:
Is there more context you can add to this, e.g. what else uses or relies on this behavior?